keyboards/cxt_studio/12e3/keyboard.json

```diff
@@ -0,0 +1,99 @@
+{
+    "manufacturer": "cxt_studio",
+    "keyboard_name": "cxt_studio 12E3",
+    "maintainer": "muge",
+    "bootloader": "atmel-dfu",
+    "diode_direction": "ROW2COL",
+    "features": {
+        "bootmagic": true,
+        "command": false,
+        "console": false,
+        "extrakey": true,
+        "mousekey": true,
+        "nkro": true,
+        "encoder": true,
+        "rgb_matrix": true,
+    },
+    "ws2812": {
+        "pin": "F7"
+    },
+    "matrix_pins": {
+        "cols": ["D4", "D7", "B4", "B5"],
+        "rows": ["C7", "C6", "D6", "F4"],
+    },
+    "encoder": {
+        "rotary": [
+            {"pin_a": "F5", "pin_b": "F6"},
+            {"pin_a": "E6", "pin_b": "F0"},
+            {"pin_a": "B3", "pin_b": "B2", "resolution": 2},
+        ]
+    },
+    "processor": "atmega32u4",
+    "rgb_matrix": {
+        "animations": {
+            "solid_color": true,
+            "breathing": true,
+            "typing_heatmap": true,
+            "digital_rain": true,
+            "solid_reactive_simple": true,
+            "solid_reactive": true,
+            "solid_reactive_wide": true,
+            "solid_reactive_multiwide": true,
+            "solid_reactive_cross": true,
+            "solid_reactive_multicross": true,
+            "solid_reactive_nexus": true,
+            "solid_reactive_multinexus": true,
+            "splash": true,
+            "multisplash": true,
+            "solid_splash": true,
+            "solid_multisplash": true
+        },
+        "default": {
+            "animation": "solid_reactive_simple"
+        },
+        "driver": "ws2812",
+        "layout": [
+            {"flags": 4, "matrix": [0, 3], "x": 3, "y": 0},
+            {"flags": 4, "matrix": [0, 2], "x": 2, "y": 0},
+            {"flags": 4, "matrix": [0, 1], "x": 1, "y": 0},
+            {"flags": 4, "matrix": [0, 0], "x": 0, "y": 0},
+            {"flags": 4, "matrix": [1, 0], "x": 0, "y": 1},
+            {"flags": 4, "matrix": [1, 1], "x": 1, "y": 1},
+            {"flags": 4, "matrix": [1, 2], "x": 2, "y": 1},
+            {"flags": 4, "matrix": [1, 3], "x": 3, "y": 1},
+            {"flags": 4, "matrix": [2, 3], "x": 3, "y": 2},
+            {"flags": 4, "matrix": [2, 2], "x": 2, "y": 2},
+            {"flags": 4, "matrix": [2, 1], "x": 1, "y": 2},
+            {"flags": 4, "matrix": [2, 0], "x": 0, "y": 2}
+        ]
+    },
+    "url": "",
+    "usb": {
+        "device_version": "1.0.0",
+        "pid": "0x12E3",
+        "vid": "0x0215"
+    },
+    "layouts": {
+        "LAYOUT": {
```

Comment:
This, and the keymap should be rearranged to better resemble the actual encoder layout.